### PR TITLE
[GNA] Update documentation for 2021.4.2

### PR DESCRIPTION
**Pull request by dorloff**
### Details:
 - Updated GNA docs for 2021.4.2

### Tickets:
 - 63610,66763

### CODE REVIEW

File: docs/IE_DG/supported_plugins/GNA.md [hunk 1]
```diff
@@ -53,8 +53,168 @@ Intel® Core™ i3-8121U Processor
...
 
+* All [12th Generation Intel® Core™ Processors (formerly codenamed Alder Lake)](https://ark.intel.com/content/www/us/en/ark/products/codename/147470/products-formerly-alder-lake.html).
+
 > **NOTE**: On platforms where Intel® GNA is not enabled in the BIOS, the driver cannot be installed, so the GNA plugin uses the software emulation mode only.
 
+### Intel® GNA Generational Differences
+
+The first and second versions of Intel® GNA found in 10th and 11th generation Intel® Core™ Processors may be considered to be functionally equivalent.  Intel® GNA 2.0 provided performance improvement with respect to Intel® GNA 1.0.  Starting with 12th Generation Intel® Core™ Processors (formerly codenamed Alder Lake), support for Intel® GNA 3.0 features is being added.
+
+In the rest of this documentation, "GNA 2.0" term will be used for GNA HW delivered with  Intel® GNA on 10th and 11th generation Intel® Core™ Processors, and "GNA 3.0" term will be used for GNA HW delivered with  Intel® GNA on 12th generation Intel® Core™ Processors.
```
Comment (mdeisher) [inline docs/IE_DG/supported_plugins/GNA.md:64]: Should say:  the term "GNA 2.0" refers to Intel(R) GNA hardware delivered on 10th and 11th generation Intel(R) Core(TM) processors, and the term "GNA 3.0" will be used to refer to GNA hardware delivered on 12th generation Intel(R) Core(TM) processors.

File: docs/IE_DG/supported_plugins/GNA.md [hunk 2]
```diff
@@ -53,8 +53,168 @@ Intel® Core™ i3-8121U Processor
...
+|7|1|1|1|1|1|384|54|54|54|54|54|36|18|
+
+
+> **NOTE**:  The above limitations only apply to the new hardware 2D convolution operation.  When possible, the Intel® GNA plugin graph compiler flattens 2D convolutions so that the second generation Intel® GNA 1D convolution operations (without these limitations) may be used.  The plugin will also flatten 2D convolutions regardless of the sizes  if GNA 2.0 compilation target is selected (see below).
+
+### Intel® GNA Forward and Backward Compatibility
+
+In the general case, there is no guarantee that a model compiled for GNA 2.0 will run on GNA 3.0, or vice versa.
+
+However, in most cases, networks compiled for GNA 2.0 will run as expected on GNA 3.0, although the performance may be worse compared to the case when a network is compiled specifically for the latter.  The exception is networks with convolutions with the number of filters greater than 8192 (see the Models and Layers Limitations section).
```
Comment (mdeisher) [inline docs/IE_DG/supported_plugins/GNA.md:212]: I don't see the Models and Layers Limitations section in this document.  Please provide a link.

File: docs/IE_DG/supported_plugins/GNA.md [hunk 3]
```diff
@@ -53,8 +53,168 @@ Intel® Core™ i3-8121U Processor
...
+
+Networks compiled for GNA 3.0 should run on GNA 2.0 with incompatible layers emulated on CPU.
+
+You can use the `KEY_GNA_EXEC_TARGET` and `KEY_GNA_COMPILE_TARGET` options  (see the Supported Configuration Parameters section below) to check interoperability.
```
Comment (mdeisher) [inline docs/IE_DG/supported_plugins/GNA.md:216]: Please provide link to Supported Configuration Parameters section.

File: docs/IE_DG/supported_plugins/GNA.md [hunk 4]
```diff
@@ -76,12 +236,13 @@ Limitations include:
 
 - Only 1D convolutions are natively supported.
 - The number of output channels for convolutions must be a multiple of 4.
+- The maximum number of filters is 65532 for GNA 3.0 and 8192 for GNA 2.0.
```
Comment (mdeisher) [inline docs/IE_DG/supported_plugins/GNA.md:239]: You have GNA2 and GNA3 reversed here.

**Review by mdeisher — CHANGES_REQUESTED** [review]

**Review by mdeisher — APPROVED** [review]

**Comment by andrew-zaytsev** [comment]
Added the changes to a new #8567

**Comment by dorloff** [comment]
Transferred to #8567.

**Pull request closed without merging**